features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Comments/CommentAdd.jsx

```diff
@@ -128,14 +133,13 @@ class CommentAdd extends React.Component {
                 .then((newComment) => {
                     this.setState({ content: '' });
                     const addedComment = newComment.body;
-                    // if (parentCommentId === null) {
-                    //     allComments.push(addedComment);
-                    // } else {
-                    //     const index = allComments.findIndex(this.filterCommentToAddReply);
-                    //     allComments[index].replies.push(addedComment);
-                    //     toggleShowReply();
-                    // }
-                    allComments.push(addedComment);
+                    if (parentCommentId === null) {
+                        allComments.push(addedComment);
+                    } else {
+                        const index = allComments.findIndex(this.filterCommentToAddReply)
+                        allComments[index].replies.list.push(addedComment);
+                        this.setState({ replyIndex: -1 });
+                    }
                     commentsUpdate(allComments);
```

Comment:
This `commentsUpdate ` call just doing the re-rendering, otherwise, we have already mutated the parent's state directly from the above lines. This is an antipattern